src/java.desktop/share/classes/javax/swing/plaf/basic/BasicTreeUI.java

```diff
@@ -1334,6 +1334,9 @@ protected void uninstallKeyboardActions() {
      */
     protected void uninstallComponents() {
         if(rendererPane != null) {
+            if (!tree.isVisible()) {
+                rendererPane.removeAll();
```

Comment:
if( -> if (

Is it actually necessary to check visibility ? 

But more importantly, in the case of the test in the bug, how and when does this uninstallComponents() method get called ? It isn't clear to me how it changes the reported behaviour of that test.

I thought uninstallComponents() is just part of uninstallUI() like when tearing down and freeing the tree, or at least switching L&F .. whereas the submitter seems to be implying nodeChanged() is the problem

Reply:
Yes, the submitter mentioned during nodeChanged() but I mentioned it before [in this comment](https://github.com/openjdk/jdk/pull/17458#issuecomment-1897684665)
that the comment present, implied it's not an issue, if the component is not removed if UI is not changed (as it is in the case when tree is not visible)...I am not sure why it was added and when but should we ignore that comment and try to remove the component even if tree is not visible (and UI is not changing)? 

I went with the thought that the coder meant what he said and I freed the components at the end (as you told during tearing down) so that there is no leak..

Reply:
My point is not about whether this is the only case we need to address, I am asking what harm could come from removing the isVisible() test here ?